### PR TITLE
Make project build endpoint return JSON

### PR DESCRIPTION
**Pull request by tetchel**
Now ~all~ more endpoints used by the IDEs return JSON.

Fixes https://github.com/eclipse/codewind/issues/2435

Signed-off-by: Tim Etchells <timetchells@ibm.com>

## What type of PR is this ? 

- [ ] Bug fix
- [x] Enhancement

## What does this PR do ?
The build endpoint seems to be the only one used by the IDEs which returns plain text instead of JSON. Also changed the error responses to provide a bit more info and use more descriptive status codes.

## Which issue(s) does this PR fix ?

https://github.com/eclipse/codewind/issues/2435

## Does this PR require a documentation change ?
I have updated the openapi.yml.

## Other notes for reviewers
I checked with Erin and John P and this change will not break Intellij or Eclipse

### CODE REVIEW

**Comment by tetchel** [comment]
@tobespc can you or a team member review?

File: src/pfe/portal/routes/projects/build.route.js [hunk 1]
```diff
@@ -35,17 +35,22 @@ async function buildProject(req, res) {
       res.status(400).send(`Unable to find project ${id}`);
       return;
     }
-    
+
     if (project.isClosed() || project.isClosing() || project.isDeleting()) {
       const msg = `Cannot perform build action ${action} on ${project.name} because it is in state ${project.state}.`;
       res.status(400).send(msg);
```
Comment (makandre) [inline src/pfe/portal/routes/projects/build.route.js]: Error 400 will still send back plain text?

Comment (tetchel) [reply src/pfe/portal/routes/projects/build.route.js]: thx

**Comment by tetchel** [comment]
Hold off, there are a couple more endpoints i'd like to update.

**Comment by tetchel** [comment]
There are too many to change; I'm going to work around it on my end, but i'd still like to put this in if we can.

**Comment by tobespc** [comment]
If we want the api's to return JSON, then we should do this across the board for consistency.  That would also mean the api's will need a version change as well

**Pull request closed without merging**